contracts/WrappedCollateralToken.sol

```diff
@@ -37,21 +37,29 @@ contract WrappedCollateralToken is IWrappedCollateralToken, ERC20Wrapper, ERC20P
         emit CapSet(newCap);
     }
 
-    function depositFor(address account, uint256 amount) public virtual override returns (bool) {
-        if (totalSupply() + amount > cap) {
-            revert ExceedsCap();
-        }
-        if (balanceOf(account) + amount > maxBalance) {
-            revert ExceedsMaxBalance();
-        }
-        return ERC20Wrapper.depositFor(account, amount);
-    }
-
     function decimals() public view virtual override(ERC20, ERC20Wrapper) returns (uint8) {
         return ERC20Wrapper.decimals();
     }
 
     function recover(address account) external override onlyOwner returns (uint256) {
         return _recover(account);
     }
+
+    function _mint(address account, uint256 amount) internal virtual override {
+        ERC20._mint(account, amount);
+        if (totalSupply() > cap) {
```

Comment:
maybe we should just do this check in `_afterTokenTransfer`?

Reply:
Every transfer would be more expensive in that case. I initially did it like that BTW

Reply:
done anyway, looks better